src/language-js/printer-estree.js

```diff
@@ -808,14 +808,21 @@ function printPathNoParens(path, options, print, args) {
       }
 
       return concat(parts);
-    case "ExportSpecifier":
+    case "ExportSpecifier": {
       parts.push(path.call(print, "local"));
 
-      if (n.exported && n.exported.name !== n.local.name) {
+      const { exported, local } = n;
+      if (
+        exported &&
+        (exported.type !== local.type ||
+          (exported.type === "Identifier" && exported.name !== local.name) ||
+          (exported.type === "StringLiteral" && exported.value !== local.value))
```

Comment:
Do we need compare `raw`?

**Prettier pr-9408**
[Playground link](https://deploy-preview-9408--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEAbOMAEAHAThmASzl0wF5MAdEACwENtsBPagbkqjgA9sJctgVEHkqUArsAAMkgGwBWAL4wiJapjoBnIXgLFcahSAA0ICNiLQNyUHVy4IAdwAKthFZR1UDukysmARrh0YADWGADK2MGEUADmyDC4YnAmcAC2-nAAJlnZADJ0cWJ0sXAAYnxpdMox8R5iMBDGtDBpqADqNITwGlFgcOFu3YQAbt1MyOAafiAxGiQwTkGxVcgAZp7zJgBWGlwAQkGhEXRpcHkxcOubKSC7XOG16ACKYhDw16hbwrbzuJP+OiZVDNPAxGDtQhZGA0ZAADkkJjwEHm7SC2EmOj+IyuJgAjm94EszO4QJoALScbLZZr4AmEfBLEqrJAbL63eZpQgJJIcp5wV7vK6sm4mGBAyHQ2FIABMYqChFQtQAwhA0iyQHANHJmmJ5gAVIHuNnfEbJACSUFysHCYFwhHMAEEreEYEx0J95goFEA)
```sh
--parser babel
```

**Input:**
```jsx
let prettier = "happy";
export { "pr\u{0065}ttier" as "prettier" }
```

**Output:**
```jsx
let prettier = "happy";
export { "pr\u{0065}ttier" };

```

Reply:
You're right. Thank you.

Btw, I'm not sure if Prettier should omit `as` when `exported` and `local` are the same.  I think it's Linter's job.

Reply:
I noticed this too, but we can fix it later?

Reply:
Raised an issue https://github.com/prettier/prettier/issues/9434

Reply:
Okay. Thank you.

Also, if we use `raw`, Prettier treats single quotes and double quotes as different things. That means

```js
// Input
export { "prettier" as 'prettier' };
export { "prettier" as "prettier" };

// Output
export { "prettier" as "prettier" };
export { "prettier" };
```

This is strange. Let's keep the current behavior as this PR and fix #9434 after merging this.